### PR TITLE
Add spec tests (WIP)

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Parameter type validation is still missing.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Thanks @Phil-Friderici !

Could you please take a look at the TODO in the class and see about removing the old sorted template. Since we are no longer supporting Ruby < 2.1.9, we shouldn't need it.

We also need a test that shows if you pass a config hash, that the correct sssd.conf is generated.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -361,7 +359,184 @@
...
+
+    it do
+      should contain_exec('authconfig-mkhomedir').with({
+        :command => '/usr/sbin/authconfig --disablesssd --disablesssdauth --update',
+        :unless  => "/usr/bin/test \"`/usr/sbin/authconfig --disablesssd --disablesssdauth --test`\" = \"`/usr/sbin/authconfig --test`\"",
+      })
+    end
+  end
+
+  describe 'with config set to valid hash' do
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb:388]: test for config hash is here

**Comment by Phil-Friderici** [comment]
regarding the template, is sorting still needed or can that simply get dropped ?

**Comment by Phil-Friderici** [comment]
Never mind, the sorting should have only been needed for consitent spec testing needs.

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (ghoneycutt) [reply spec/classes/init_spec.rb:388]: aha! thanks!

**Comment by ghoneycutt** [comment]
yeah, the sorting can be dropped since we dont support old ruby versions. We can also drop :rubyversion from the tests

**Pull request merged**